clients/hadoopfs/src/main/java/io/lakefs/LakeFSFileSystem.java

```diff
@@ -241,7 +240,15 @@ public FSDataOutputStream append(Path path, int i, Progressable progressable) th
     /**
      * This method is implemented under the following assumptions:
      * 1. rename is only supported for uncommitted data on the same branch.
-     * 2. file rename operation is supported, directories rename is unsupported.
+     * 2. The following rename scenarios are supported:
+     *      * file -> existing-file-name: rename(src.txt, existing-dst.txt) -> existing-dst.txt, existing-dst.txt is overriden
+     *      * file -> existing-directory-name: rename(src.txt, existing-dstdir) -> existing-dstdir/src.txt
```

Comment:
[nit, not in this PR] Maybe explain here (or somewhere...) what's a "directory"?